packages/builder/src/helpers/validation/yup/index.js

```diff
@@ -0,0 +1,58 @@
+import { capitalise } from "helpers"
+import { object } from "yup"
+import { writable, get } from "svelte/store"
+
+export const createValidationStore = () => {
+  const DEFAULT = {
+    errors: {},
+    touched: {},
+    valid: false,
+  }
+
+  const validator = {}
+  const validation = writable(DEFAULT)
+
+  const addValidator = (propertyName, propertyValidator) => {
+    if (!propertyValidator || !propertyName) {
+      return
+    }
+    validator[propertyName] = propertyValidator
+  }
+
+  const check = async values => {
+    const obj = object().shape(validator)
+    // clear the previous errors
+    const properties = Object.keys(validator)
+    properties.forEach(property => (get(validation).errors[property] = null))
+    try {
+      await obj.validate(values, { abortEarly: false })
+    } catch (error) {
+      error.inner.forEach(err => {
```

Comment:
Good shout. To address this I'm going to add a notification
```js
  if (!error.inner) {
    notifications.error("Unexpected validation error", error)
  }
```
so that we can give some feedback to the user that something is not right, and setting the form to `valid = false` so that submit will be prevented.